docs/docs/llms/MLflow Tracing/introduction.mdx

```diff
@@ -0,0 +1,1146 @@
+    mlflow.openai.autolog()
+
+    mlflow.set_experiment("OpenAI Swarm")
+
+    client = Swarm()
+
+
+    def transfer_to_agent_b():
+        return agent_b
+
+
+    agent_a = Agent(
+        name="Agent A",
+        instructions="You are a helpful agent.",
+        functions=[transfer_to_agent_b],
+    )
+
+    agent_b = Agent(
+        name="Agent B",
+        instructions="Only speak in Haikus.",
+    )
+
+    response = client.run(
+        agent=agent_a,
+        messages=[{"role": "user", "content": "I want to talk to agent B."}],
+    )
+    print(response)
+    ```
+
+    The logged trace, associated with the `OpenAI Swarm` experiment, can be seen in the MLflow UI, as shown below:
+
+    ![OpenAI Swarm Tracing](/images/llms/tracing/openai-swarm-tracing.png)
+
+  </TabItem>
+  <TabItem value="llamaindex" label="LlamaIndex">
+    ### LlamaIndex Automatic Tracing
+
+    The MLflow LlamaIndex flavor's autologging feature has a direct integration with MLflow tracing. When LlamaIndex autologging is enabled with <APILink fn="mlflow.llama_index.autolog" />, invocation of components
+    such as LLMs, agents, and query/chat engines will automatically record generated traces during interactive development.
+
+    ```python
+    import mlflow
+
+    mlflow.llama_index.autolog()
+    ```
+
+    To see the full example of tracing LlamaIndex, please visit [LLamaIndex Tracing documentation](_).
+
+    ![LlamaIndex Tracing](/images/llms/llama-index/llama-index-trace.png)
+
+  </TabItem>
+  <TabItem value="autogen" label="AutoGen">
+    ### AutoGen Automatic Tracing
+
+    MLflow Tracing ensures observability for your AutoGen application that involves complex multi-agent interactions. You can enable auto-tracing by calling <APILink fn ="mlflow.autogen.autolog" />,
+    then the internal steps of the agents chat session will be logged to the active MLflow Experiment.
+
+    ```python
+    import mlflow
+
+    mlflow.autogen.autolog()
+    ```
+
+    To see the full example of tracing AutoGen, please refer to the [AutoGen Tracing example](https://github.com/mlflow/mlflow/tree/master/examples/autogen/tracing.py).
+
+    ![AutoGen Tracing](/images/llms/autogen/autogen-trace.png)
+
+  </TabItem>
+</Tabs>
+
+## Tracing Fluent APIs
+
+MLflow's <APILink fn="mlflow.start_span">`fluent APIs`</APILink> provide a straightforward way to add tracing to your functions and code blocks.
+By using decorators, function wrappers, and context managers, you can easily capture detailed trace data with minimal code changes.
+
+As a comparison between the fluent and the client APIs for tracing, the figure below illustrates the differences in complexity between the two APIs,
+with the fluent API being more concise and the recommended approach if your tracing use case can support using the higher-level APIs.
+
+<img
+  alt="Fluent vs Client APIs"
+  class="center"
+  src="/images/llms/tracing/fluent-vs-client-tracing.png"
+  style={{ width: "60%" }}
+/>
```

Comment:
Same comment of images here, in order to make sure the `src`s don't break, let's use markdown syntax and apply styling by wrapping it in a `div`:

```suggestion
<div class="center" style={{ width: "60%" }}>
  ![Fluent vs Client APIs](/images/llms/tracing/fluent-vs-client-tracing.png)
</div>
```